pkg/cmd/get/clusters.go

```diff
@@ -21,14 +67,313 @@ func preClustersE(cmd *cobra.Command, args []string) error {
+		return nil, err
+	}
+
+	// Create an empty array of clusters to collect the information
+	clusterList := []Cluster{}
+
+	// List the idp builder clusters according to the provider: podman or docker
+	provider := cluster.NewProvider(cluster.ProviderWithLogger(kind.KindLoggerFromLogr(&logger)), detectOpt)
 	clusters, err := provider.List()
 	if err != nil {
-		return fmt.Errorf("failed to list clusters: %w", err)
+		//logger.Error(err, "failed to list clusters.")
+		return nil, err
+	}
+
+	// Populate a list of Kube client for each cluster/context matching a idpbuilder cluster
+	manager, _ := CreateKubeClientForEachIDPCluster(config, clusters)
+
+	for _, cluster := range clusters {
+		aCluster := Cluster{Name: cluster}
+
+		// Search about the idp cluster within the kubeconfig file and show information
+		c, found := findClusterByName(config, "kind-"+cluster)
+		if !found {
+			//logger.Error(nil, fmt.Sprintf("Cluster not found: %s within kube config file\n", cluster))
+			return nil, err
+		} else {
+			cli, err := GetClientForCluster(manager, cluster)
+			if err != nil {
+				//logger.Error(err, fmt.Sprintf("failed to get the context for the cluster: %s.", cluster))
+				return nil, err
+			}
+			logger.V(1).Info(fmt.Sprintf("Got the context for the cluster: %s.", cluster))
+
+			// Print the external port mounted on the container and available also as ingress host port
+			targetPort, err := findExternalHTTPSPort(cli)
+			if err != nil {
+				//logger.Error(err, "failed to get the kubernetes ingress service.")
+				return nil, err
+			} else {
+				aCluster.ExternalPort = targetPort
+			}
+
+			aCluster.URLKubeApi = c.Server
+			aCluster.TlsCheck = c.InsecureSkipTLSVerify
+
+			// Print the internal port running the Kuber API service
+			kubeApiPort, err := findInternalKubeApiPort(cli)
+			if err != nil {
+				//logger.Error(err, "failed to get the kubernetes default service.")
+				return nil, err
+			} else {
+				aCluster.KubePort = kubeApiPort
+			}
+
+			// Let's check what the current node reports
+			var nodeList corev1.NodeList
+			err = cli.List(context.TODO(), &nodeList)
+			if err != nil {
+				//logger.Error(err, "failed to list nodes for the current kube cluster.")
+				return nil, err
+			}
+
+			for _, node := range nodeList.Items {
+				nodeName := node.Name
+
+				aNode := Node{}
+				aNode.Name = nodeName
+
+				for _, addr := range node.Status.Addresses {
+					switch addr.Type {
+					case corev1.NodeInternalIP:
+						aNode.InternalIP = addr.Address
+					case corev1.NodeExternalIP:
+						aNode.ExternalIP = addr.Address
+					}
+				}
+
+				// Get Node capacity
+				aNode.Capacity = populateNodeCapacity(node.Status.Capacity)
+
+				// Get Node Allocated resources
+				allocated, err := printAllocatedResources(context.Background(), cli, node.Name)
+				if err != nil {
+					//logger.Error(err, "failed to get the allocated resources.")
+					return nil, err
+				}
+				aNode.Allocated = allocated
```

Comment:
Are we doing anything with `aNode`? Looks like it's discarded each iteration?

Reply:
Good question. Do we want part of the command "get clusters" to also show the allocated or available resources OR this is something that we will only show when we will display the detail of a cluster ?

Reply:
Hmm it would be nice to have but I'd say it's not critical since you can get that information with `kubectl describe nodes` command.

Reply:
The big plus of our command is that we could show everything with one command (= developer like that) in one place vs running several kubectl commands ;-)

Reply:
I don't see `aNode` value used for anything here still.

Reply:
Added a new column to list the nodes 

```
NAME         EXTERNAL-PORT   KUBE-API                  TLS     KUBE-PORT   NODES
my-konflux   8443            https://127.0.0.1:57063   false   6443        my-konflux-control-plane
```

Commit: 9fc54e2

Reply:
See: 9fc54e2